src/fhirtypes/common.ts

```diff
@@ -150,8 +151,9 @@ export function setPropertyOnInstance(
           }
           const sliceIndices: number[] = [];
           // Find the indices where slices are placed
+          const sliceExtensionUrl = fisher.fishForMetadata(sliceName)?.url;
           current[pathPart.base]?.forEach((el: any, i: number) => {
-            if (el?._sliceName === sliceName) {
+            if (el?._sliceName === sliceName || (el?.url && el?.url === sliceExtensionUrl)) {
```

Comment:
The addition of this makes sense to me! Is there a way to add a test for this case? Or is it fairly difficult to do? Or is it already covered and I missed it?

Reply:
I thought I had tested this, but looking at the tests, I guess I must not have. I'll add some tests.

Reply:
Done in https://github.com/FHIR/sushi/pull/772/commits/cbfb200312b54f54161ee000ac050c1c964d32db.